core/workflow/loadDPL.go

```diff
@@ -36,20 +36,32 @@ func LoadDPL(tasks []*task.Class) (workflow Role, err error) {
 		SingleTaskRole := taskRole{
 			roleBase: roleBase{
 				Name:        taskItem.Identifier.Name,
+				parent:      root,
 				Connect:     nil,
 				Constraints: nil,
 				Defaults:    gera.MakeStringMap(),
 				Vars:        gera.MakeStringMap(),
+				UserVars:    gera.MakeStringMap(),
+				Locals:      nil,
 				Bind:        nil,
 			},
 		}
 
-		SingleTaskRole.Connect = taskItem.Connect
-		SingleTaskRole.Constraints = taskItem.Constraints
+		for _, eachConnect := range taskItem.Connect {
+			SingleTaskRole.Connect = append(SingleTaskRole.Connect, eachConnect)
+		}
+
+		for _, eachConstraint := range taskItem.Constraints {
+			SingleTaskRole.Constraints = append(SingleTaskRole.Constraints, eachConstraint)
+		}
```

Comment:
Your approach is correct for a manual deep copy, but in Go you can use a simpler single statement with the `...` operator to replace the whole loop:
`SingleTaskRole.Connect = append(SingleTaskRole.Connect, taskItem.Connect...)`

This improves readability because one can immediately tell it's a straight deep copy without in-loop manipulation of each entry.

Reply:
Perfect! That looks much better